app/src/components/GrantCard.vue

```diff
@@ -0,0 +1,65 @@
+<!-- GrantCard -->
+
+<template>
+  <figure class="group">
+    <div class="relative cursor-pointer" @click="pushRoute({ name: 'dgrants-id', params: { id: id.toString() } })">
+      <img class="shadow-light" :src="imgurl" />
+      <div class="absolute bottom-0 right-0">
+        <!-- when item in cart, add class "in-cart" to <button> -->
+        <button
+          class="btn opacity-0 group-hover:opacity-100"
+          :class="{ 'in-cart': isInCart(id) }"
+          @click.stop="addToCart(id)"
+        >
+          <CartIcon />
+        </button>
+      </div>
+    </div>
+    <figcaption class="mt-4">
+      <div class="text-grey-500 font-medium truncate">{{ name }}</div>
+      <div class="flex justify-between">
+        <span class="text-grey-400"
+          >by
+          <a
+            class="text-grey-500 underline"
+            :href="getEtherscanUrl(ownerAddress, 1, 'address')"
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            {{ formatAddress(ownerAddress) }}
+          </a>
+        </span>
+        <span class="text-grey-400"
+          >Raised: <span class="text-grey-500">{{ raised }}</span></span
+        >
+      </div>
+    </figcaption>
+  </figure>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+// --- Store ---
+import useCartStore from 'src/store/cart';
+// --- Methods and Data ---
+import { BigNumber } from 'ethers';
+import { formatAddress, getEtherscanUrl, pushRoute } from 'src/utils/utils';
+// --- Icons ---
+import { Cart2Icon as CartIcon } from '@fusion-icons/vue/interface';
+
+export default defineComponent({
+  name: 'GrantCard',
+  props: {
+    id: { type: BigNumber, required: true },
```

Comment:
Let's use `BigNumberish` as the type and then when using the `id` you can do `BigNumber.from(id).toString()`, etc. In general you shouldn't use `BigNumber` as an input type because it's an ethers specific type, but you can/should:
- use `BigNumberish` as an input type and handle the input type
- use `BigNumber` as a return type (not applicable here, but worth mentioning)

Reply:
Got it! 

When using it, it seems that `id.toString()` works fine. What is the goal of using `BigNumber.from(id).toString()`? (I changed it anyway)

Reply:
You are right that numbers, strings, BigInt, and BigNumber all have a `.toString()` method, so `id.toString()` will pretty much always work. Using `BigNumber.from(id).toString()` instead basically forces the value to be an integer, since `BigNumber.from(id)` will throw if you pass in decimals. For this use case either approach is probably ok